internal/oidc/token/token_handler.go

```diff
@@ -320,6 +343,14 @@ func upstreamLDAPRefresh(ctx context.Context, providerCache oidc.UpstreamIdentit
 	// Replace the old value with the new value.
 	session.Fosite.Claims.Extra[oidc.DownstreamGroupsClaim] = groups
 
+	added, removed := diffSortedGroups(oldGroups, groups)
+	if added != "" {
+		warning.AddWarning(ctx, "", "groups have been added since login: "+added)
+	}
+	if removed != "" {
+		warning.AddWarning(ctx, "", "groups have been removed since login: "+removed)
+	}
```

Comment:
Can we implement the warning in a generic way / at a high level so that we do not need to remember to add them for each new upstream refresh?